File: backend/src/models/Lead.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const Lead = sequelize.define('Lead', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
  phone: {
    type: DataTypes.STRING(20),
    allowNull: false,
    unique: true,
    comment: 'Unique mobile number - core identifier',
  },
  name: {
    type: DataTypes.STRING(150),
    allowNull: false,
  },
  company_name: {
    type: DataTypes.STRING(200),
  },
  email: {
    type: DataTypes.STRING(150),
    validate: { isEmail: true },
  },
  status: {
    type: DataTypes.ENUM(
      'new',           // جديد
      'contacted',     // تم الاتصال
      'follow_up',     // متابعة
      'interested',    // مهتم
      'not_interested',// غير مهتم
      'negotiating',   // قيد التفاوض
      'qualified',     // مرشح للتعاقد
      'contracted',    // تم التعاقد
      'rejected'       // مرفوض
    ),
    defaultValue: 'new',
  },
  current_level: {
    type: DataTypes.INTEGER,
    defaultValue: 1,
    comment: '1=Sales, 2=Manager, 3=Admin',
  },
  first_contact_date: {
    type: DataTypes.DATEONLY,
  },
  follow_up_date: {
    type: DataTypes.DATEONLY,
    comment: 'Next follow-up date set by Level 1',
  },
  meeting_date: {
    type: DataTypes.DATE,
    comment: 'Meeting date set by Level 2',
  },
  notes: {
    type: DataTypes.TEXT,
  },
  source: {
    type: DataTypes.STRING(100),
    comment: 'Lead source: social_media, referral, website, etc.',
  },
  assigned_to: {
    type: DataTypes.INTEGER,
    comment: 'Assigned sales user (Level 1)',
  },
  manager_id: {
    type: DataTypes.INTEGER,
    comment: 'Assigned manager (Level 2)',
  },
  created_by: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  transferred_at: {
    type: DataTypes.DATE,
    comment: 'When lead was transferred to Level 2',
  },
  contracted_value: {
    type: DataTypes.DECIMAL(15, 2),
    comment: 'Contract value if contracted',
  },
  priority: {
    type: DataTypes.ENUM('low', 'medium', 'high'),
    defaultValue: 'medium',
  },
}, {
  tableName: 'leads',
  indexes: [
    { fields: ['phone'] },
    { fields: ['status'] },
    { fields: ['assigned_to'] },
    { fields: ['follow_up_date'] },
  ],
});

module.exports = Lead;

File: mobile/lib/models/lead.dart
class LeadActivity {
  final int id;
  final String type;
  final String description;
  final String? callResult;
  final String? userName;
  final DateTime createdAt;

  LeadActivity({
    required this.id,
    required this.type,
    required this.description,
    this.callResult,
    this.userName,
    required this.createdAt,
  });

  factory LeadActivity.fromJson(Map<String, dynamic> json) => LeadActivity(
        id: json['id'],
        type: json['type'],
        description: json['description'],
        callResult: json['call_result'],
        userName: json['user']?['name'],
        createdAt: DateTime.parse(json['created_at']),
      );
}

class Lead {
  final int id;
  final String phone;
  final String name;
  final String? companyName;
  final String? email;
  final String status;
  final String priority;
  final String? notes;
  final String? source;
  final DateTime? followUpDate;
  final String? assigneeName;
  final String? managerName;
  final List<LeadActivity> activities;
  final int currentLevel;

  Lead({
    required this.id,
    required this.phone,
    required this.name,
    this.companyName,
    this.email,
    required this.status,
    required this.priority,
    this.notes,
    this.source,
    this.followUpDate,
    this.assigneeName,
    this.managerName,
    this.activities = const [],
    this.currentLevel = 1,
  });

  factory Lead.fromJson(Map<String, dynamic> json) => Lead(
        id: json['id'],
        phone: json['phone'],
        name: json['name'],
        companyName: json['company_name'],
        email: json['email'],
        status: json['status'] ?? 'new',
        priority: json['priority'] ?? 'medium',
        notes: json['notes'],
        source: json['source'],
        followUpDate: json['follow_up_date'] != null
            ? DateTime.tryParse(json['follow_up_date'])
            : null,
        assigneeName: json['assignee']?['name'],
        managerName: json['manager']?['name'],
        currentLevel: json['current_level'] ?? 1,
        activities: (json['activities'] as List<dynamic>? ?? [])
            .map((a) => LeadActivity.fromJson(a))
            .toList(),
      );
}

